### PR TITLE
Data Structures and Algorithms Course in Java,C++,Python and Javascript

### PR DESCRIPTION
**Pull request by ummeq**
## What does this PR do?
Add Free DSA course

### Why is this valuable (or not)?
This is 𝒇𝒓𝒆𝒆 𝒄𝒆𝒓𝒕𝒊𝒇𝒊𝒄𝒂𝒕𝒆 from 𝐆𝐨𝐨𝐠𝐥𝐞 by learning 𝐃𝐚𝐭𝐚 𝐒𝐭𝐫𝐮𝐜𝐭𝐮𝐫𝐞𝐬 𝐚𝐧𝐝 𝐀𝐥𝐠𝐨𝐫𝐢𝐭𝐡𝐦𝐬 with free practice material.
The course is available in four programming languages (Javascript, Java, C++ and Python)

### How do we know it's really free?
Yes this course by Google and free

### For book lists, is it a book? For course lists, is it a course? etc.
Course 

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by ummeq** [comment]
Hello @eshellman 

Request you please review and merge the PR

**Comment by ummeq** [comment]
Hello @eshellman 

Request you please review and merge the PR

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -141,6 +141,7 @@
 * [Data Structures and Algorithms for Beginners](https://www.youtube.com/watch?v=BBpAmxU_NQo) - Mosh Hamedani (Programming with Mosh)
 * [Data Structures and Algorithms Full Course](https://www.youtube.com/watch?v=LcKqYoX8uH4) - Simplilearn (YouTube, Video length 7:13:56)
 * [Data Structures and Algorithms in Java Full Course](https://youtube.com/playlist?list=PL6Zs6LgrJj3tDXv8a_elC6eT_4R5gfX4d) - Dinesh Varyani (YouTube playlist)
+* [Data Structures and Algorithms in Java,C++,Python and Javascript) ](https://techdevguide.withgoogle.com/paths/data-structures-and-algorithms/?no-filter=true) - Tech Dev Guide by Google
```
Comment (eshellman) [inline courses/free-courses-en.md:144]:
```suggestion
* [Data Structures and Algorithms](https://techdevguide.withgoogle.com/paths/data-structures-and-algorithms/) - Tech Dev Guide by Google
```
Don't invent titles! Use the title given by the resource.

**Comment by ummeq** [comment]
Hello @eshellman

I have update the PR .

Request you please review and merge the PR

**Comment by ummeq** [comment]
Hello @eshellman

I have update the PR .

Request you please review and merge the PR

**Comment by ummeq** [comment]
Hello @eshellman

I have update the PR .

Request you please review and merge the PR

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**